add_gist.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'rubygems'
+require 'net/http'
+require 'uri'
+require 'json'
+require 'OpenSSL'
+require 'find'
+
+class PathNotFoundError < StandardError; end
+
+def read_files(pwd = Dir.pwd)
+  files = {}
+  begin
+    unless File.exist?(pwd)
+      raise PathNotFoundError, 'Path to file or directory not found'
+    end
+    Find.find(pwd) do |path|
+      if File.file?(path)
+        files.merge!(File.basename(path) => { 'content': File.open(path, 'r+').read })
+      end
+    end
+  rescue PathNotFoundError => e
+    puts e.message
+  end
+  files
+end
+
+def read_access_token
+  file = File.open('./accessToken.env', 'r+')
+  JSON.parse(file.read)
+end
+
+def send_request(files, description, is_public, token)
+  uri = URI.parse('https://api.github.com/gists')
+  uri.query = URI.encode_www_form(access_token: token['access_token'])
+
+  http = Net::HTTP.new(uri.host, uri.port)
+  http.use_ssl = (uri.scheme == 'https')
+  http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+
+  request = Net::HTTP::Post.new(uri.request_uri)
+  request.body = {
+    "description": description.to_s,
+    "public": is_public,
+    "files": files
+  }.to_json
+
+  http.request(request)
+end
+
+path = ARGV[0]
```

Comment:
I don't like to see all this code dropped here, can we have this wrapped in a method?